yt_dlp/extractor/porndead.py

```diff
@@ -0,0 +1,127 @@
+import re
+import urllib.parse
+
+from yt_dlp.utils._utils import int_or_none
+
+from .common import InfoExtractor
+from ..utils import ExtractorError
+
+
+class PornDeadIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?porndead\.org/video/(?P<id>[0-9a-f]+)'
+    _TESTS = [
+        {
+            'url': 'https://porndead.org/video/65fefcb523810',
+            'info_dict': {
+                'id': '65fefcb523810',
+                'ext': 'mp4',
+                'title': 'Hysterical Literature - Isabel Love',
+                'age_limit': 18,
+            },
+        },
+    ]
+
+    def _real_extract(self, url):
+        url = url.strip().lower()
+
+        # if www is missing, add it because the relative URLs seem to depend on it
+        parsed = urllib.parse.urlparse(url)
+        if parsed.netloc == 'porndead.org':
+            parsed = parsed._replace(netloc='www.porndead.org')
+            url = urllib.parse.urlunparse(parsed)
+
+        video_id = self._match_id(url)
```

Comment:
Since we're matching the URL, we can use that instead of parsing the URL (which isn't wrong, obvs):
 ```suggestion
        video_id, www = self._match_valid_url(url).group('id', 'www')
        # if www is missing, add it because the relative URLs seem to depend on it
        if not www:
            url = url.replace('://', '://www.')
```